pov_tools/create_POV_data_files

```diff
@@ -0,0 +1,218 @@
+		for strip in ${source[@]}; do
+			sourcename="${strip##*/}"
+			sourcebase="${sourcename%.*}"
+		done
+		echo "Found the following source image file: ${sourcename[*]}"
+		echo ""
+		# check for multiples and for filename contains 'preview'
+		recheck=$(ls 2>/dev/null -Ubad1 -- *.png | wc -l)
+		if [[ $sourcename == *"preview"* ]]; then
+			echo "OOPS! Source file name can not contain 'preview'."
+			echo "Please use a different file name."
+			echo "Press q to quit, or"
+			read -n 1 -r -s -p $'press any key to try again.\n' retry
+			if [ "$retry" = "q" ]; then exit; fi
+			exec "./create_POV_data_files"
+		elif [ $recheck -ge 2 ]; then
+			exec "./create_POV_data_files"
+		fi
+		image="$sourcename";;
+esac
+
+valid_option=0
+
+while [[ "$valid_option" != 1 ]]
+do
+    echo "Please select the settings you want to create the POV image with: "
+    echo "1 - Preset 1 (HEIGHT=144, LENGTH=39, OFFSET=18)"
+    echo "2 - Preset 2 (HEIGHT=97,  LENGTH=63, OFFSET=50)"
+    echo "3 - Preset 3 (HEIGHT=72,  LENGTH=50, OFFSET=15)"
+    echo "4 - Preset 4 (HEIGHT=72,  LENGTH=50, OFFSET=39)"
+    echo "5 - Height entry only"
+    echo "6 - Custom options"
+    echo "q - to quit"
+	read option
+	case $option in
+		"")	echo "! Some option is needed. Please try again"; echo "";;
+		1)	make IMAGE=$image OPTIONS="--height=144 --length=39 --offset=18"; HEIGHT=144 preset=1 valid_option=1;;
+		2)	make IMAGE=$image OPTIONS="--height=97 --length=63 --offset=50"; HEIGHT=97 preset=1 valid_option=1;;
+		3)	make IMAGE=$image OPTIONS="--height=72 --length=50 --offset=15"; HEIGHT=72 preset=1 valid_option=1;;
+		4)  make IMAGE=$image OPTIONS="--height=72 --length=50 --offset=39"; HEIGHT=72 preset=1 valid_option=1;;
+		5)  height_only=1; valid_option=1;;
+		6)  valid_option=1;;
+#		7) ------ ADDITIONAL USER PRESETS CAN BE ADDED HERE LIKE THIS-------
+#		8) make IMAGE=$image OPTIONS="--height=H --length=L --offset=O"; HEIGHT=H preset=1 valid_option=1;;
+#		9)		 or move 4 & 5 down to keep them all together
+#       *) H is amount of pixels in your blade,
+#       *) L is the length of that blade in the unit of the distance of your choosing (most common inches or centimeters)
+#       *) O is offset - the distance (same unit as L) from the bottom of the blade to the fulcrum (e.g. the joint you are swinging with)
+		*[q*]*)	exit;;
+		*)	echo ""
+			echo "! Not a valid option, Please try again,"
+			echo "or type q to quit. "; echo "";;
+	esac
+done
+
+validinput=0
+
+while [[ "$validinput" != 1 ]] && [[ $preset == 0 ]]
+do
+	echo ""
+	echo "How many pixels are in your blade? (Press ENTER for default 144) "
+	read  height
+	if [[ $height == "" ]]; then height=144 length=39 offset=18 validinput=1; echo "Default 144"
+	elif [[ $height == "q" ]]; then exit
+	elif [[ $height -gt 264 ]] && [[ $height -le 576 ]]; then
+		read -p "Are you sure there's that many pixels?? (y/n)" too_many
+		if [[ $too_many == "y" ]]; then validinput=1; else echo ""; fi
+	elif [[ $height -gt 576 ]]; then
```

Comment:
If you check -gt 576 first, then you don't have to check it again like you do on line 129

Reply:
Still need to check both ranges though as much as I can figure... see alteration.